hr_holidays_weekend/hr_holidays.py

```diff
@@ -0,0 +1,213 @@
+        holidays = []
+
+        employee = self.pool.get('hr.employee').browse(cr, uid, employee_id)
+
+        hr_holiday_public_obj = self.pool.get('hr.holidays.public')
+        if employee.address_id.country_id is None:
+            country_filter = [('country_id', '=', False)]
+        else:
+            country_filter = ['|', ('country_id', '=',
+                                    employee.address_id.country_id.id),
+                                   ('country_id', '=', False)]
+        hr_holiday_public = hr_holiday_public_obj.search(cr, uid,
+                                                         country_filter)
+
+        if employee.address_id.state_id is None:
+            states_filter = [('holidays_id', 'in',
+                              hr_holiday_public),
+                             ('state_ids', '=', False)]
+        else:
+            states_filter = [('holidays_id', 'in',
+                              hr_holiday_public),
+                             '|', ('state_ids.id', '=',
+                                   employee.address_id.state_id.id),
+                             ('state_ids', '=', False)]
+
+        hr_holiday_public_line_obj = self.pool.get('hr.holidays.public.line')
+        holidays_line_ids = hr_holiday_public_line_obj.search(cr, uid,
+                                                              states_filter)
+        for holiday_line_id in holidays_line_ids:
+            line = hr_holiday_public_line_obj.browse(cr, uid, holiday_line_id)
+            holidays.append(line.date)
+        return holidays
+
+    def onchange_date_from(self, cr, uid, ids, date_to, date_from,
+                           employee_id):
+        """
+        If there are no date set for date_to, automatically
+        set one 8 hours later than
+        the date_from.
+        Also update the number_of_days.
+        """
+
+        # date_to has to be greater than date_from
+        if (date_from and date_to) and (date_from > date_to):
+            raise orm.except_orm(_('Warning!'), _("""The start date must be
+                anterior to the end date."""))
+
+        result = {'value': {}}
+
+        # No date_to set so far: automatically compute one 8 hours later
+        if date_from and not date_to:
+            date_to_with_delta =\
+                datetime.datetime.strptime(date_from,
+                                           DEFAULT_SERVER_DATETIME_FORMAT) +\
+                datetime.timedelta(hours=8)
+            result['value']['date_to'] = str(date_to_with_delta)
+
+        # new timezone-shift
+        user_obj = self.pool.get("res.users").browse(cr, uid, uid)
+
+        # Compute and update the number of days
+        if (date_to and date_from) and (date_from <= date_to):
+            diff_day = self._get_number_of_days(cr, uid, date_from, date_to,
+                                                user_obj.tz,
+                                                user_obj.tz_offset,
```

Comment:
You can use the context to retreive the time zone.
tz_offset is not used in your function _get_number_of_days, you may remove it.